pkg/workceptor/workceptor.go

```diff
@@ -689,6 +689,31 @@ func (w *Workceptor) CancelUnit(unitID string) error {
 		return fmt.Errorf("unknown work unit %s", unitID)
 	}
 	w.activeUnitsLock.RUnlock()
+	remoteUnitID := unit.status.RemoteUnitID
```

Comment:
We should get a lock on `unit.lock` here.  (This was my error, we should have already had a lock before manipulating the unit status at the bottom of the function.)